istio/tests/test_e2e.py

```diff
@@ -9,11 +9,22 @@
 from .common import CITADEL_METRICS, GALLEY_METRICS, MESH_METRICS, NEW_MIXER_METRICS, PILOT_METRICS
 
 
+INTERMITTENT_METRICS = [
+    'istio.mesh.request.count',
+    'istio.pilot.mcp_sink.recv_failures_total',
+]
+
+
 @pytest.mark.e2e
 def test_e2e(dd_agent_check):
     aggregator = dd_agent_check(rate=True)
     for metric in MESH_METRICS + NEW_MIXER_METRICS + GALLEY_METRICS + PILOT_METRICS + CITADEL_METRICS:
-        aggregator.assert_metric(metric)
+        try:
+            aggregator.assert_metric(metric)
+        except AssertionError:
+            # Filter those out so that we can keep the assert_all_metrics_covered
+            if metric not in INTERMITTENT_METRICS:
+                raise
```

Comment:
IHMO would suggest something like this to avoid the try/except/raise:

```
    for metric in MESH_METRICS + NEW_MIXER_METRICS + GALLEY_METRICS + PILOT_METRICS + CITADEL_METRICS: # without the two intermittent metrics 
        aggregator.assert_metric(metric)

    for metric in INTERMITTENT_METRICS:
        aggregator.assert_metric(metric, at_least=0)
```

Reply:
The constants are used elsewhere though...

Reply:
Then maybe this ?

```
    for metric in MESH_METRICS + NEW_MIXER_METRICS + GALLEY_METRICS + PILOT_METRICS + CITADEL_METRICS:
        if metric in INTERMITTENT_METRICS:
            aggregator.assert_metric(metric, at_least=0)
        else:
            aggregator.assert_metric(metric)

```